code/game/objects/structures/xeno.dm

```diff
@@ -426,9 +426,42 @@
 			hugger.go_idle(TRUE)
 	addtimer(CALLBACK(src, .proc/Grow), rand(EGG_MIN_GROWTH_TIME, EGG_MAX_GROWTH_TIME))
 
-/obj/effect/alien/egg/Destroy()
-	QDEL_LIST(egg_triggers)
-	return ..()
+	var/list/turf/target_locations = filled_turfs(src, trigger_size, "circle", FALSE)
+	for(var/i in target_locations)
+		RegisterSignal(i, COMSIG_ATOM_ENTERED, .proc/egg_proximity)
+
+
+///Triggers the egg when hostiles get too close
+/obj/effect/alien/egg/proc/egg_proximity(datum/source, atom/hostile)
+	SIGNAL_HANDLER
+
+	if(status != EGG_GROWN || status == (EGG_BURST|EGG_DESTROYED) ) //Don't bother if we're not grown or we're already burst/destroyed
```

Comment:
redundant check, egg_burst and egg_destroyed are different from egg_grown so this evaluation doenst need to happen
```suggestion
	if(status != EGG_GROWN) //Don't bother if we're not grown or we're already burst/destroyed
```